tools/rum-models-generator/Sources/RUMModelsGeneratorCore/Input/ObjcInterop/ObjcInteropTypeReader.swift

```diff
@@ -0,0 +1,197 @@
+                    return propertyWrapper
+                case let swifTypeReference as SwiftTypeReference:
+                    let referencedType = try resolve(swiftTypeReference: swifTypeReference)
+
+                    switch referencedType {
+                    case let swiftStruct as SwiftStruct:
+                        let propertyWrapper = ObjcInteropPropertyWrapperAccessingNestedStruct(
+                            owner: objcClass,
+                            swiftProperty: swiftProperty
+                        )
+                        propertyWrapper.objcNestedClass = ObjcInteropReferencedTransitiveClass(
+                            owner: propertyWrapper,
+                            managedSwiftStruct: swiftStruct
+                        )
+                        return propertyWrapper
+                    case let swiftEnum as SwiftEnum:
+                        let propertyWrapper = ObjcInteropPropertyWrapperAccessingNestedEnum(
+                            owner: objcClass,
+                            swiftProperty: swiftProperty
+                        )
+                        propertyWrapper.objcNestedEnum = ObjcInteropReferencedEnum(
+                            owner: propertyWrapper,
+                            managedSwiftEnum: swiftEnum
+                        )
+                        return propertyWrapper
+                    default:
+                        throw Exception.illegal("Illegal reference type: \(swifTypeReference)")
+                    }
+                default:
+                    throw Exception.unimplemented(
+                        "Cannot generate @objc property wrapper for: \(type(of: swiftProperty.type))"
+                    )
+                }
+            }
+
+        try objcClass.objcPropertyWrappers
+            .compactMap { $0 as? ObjcInteropPropertyWrapperForTransitiveType }
+            .forEach { propertyWrapper in
+                // Store `ObjcInteropTypes` created for property wrappers
+                outputObjcInteropTypes.append(propertyWrapper.objcTransitiveType)
+                if let transitiveClass = propertyWrapper.objcTransitiveType as? ObjcInteropClass {
+                    // Recursively generate property wrappers for each transitive `ObjcInteropClass`
+                    try generateTransitiveObjcInteropTypes(in: transitiveClass)
+                }
+            }
+    }
+
+    private func objcInteropType(for swiftType: SwiftType) throws -> ObjcInteropType {
+        switch swiftType {
+        case _ as SwiftPrimitive<Bool>,
+             _ as SwiftPrimitive<Double>,
+             _ as SwiftPrimitive<Int>,
+             _ as SwiftPrimitive<Int64>:
+            return ObjcInteropNSNumber(swiftType: swiftType)
+        case let swiftString as SwiftPrimitive<String>:
+            return ObjcInteropNSString(swiftString: swiftString)
+        case let swiftArray as SwiftArray:
```

Comment:
Another naive question, don't we have a use for `Set`/`Dictionary` as well in our public APIs?

Reply:
Similar to [this](https://github.com/DataDog/dd-sdk-ios/pull/382#discussion_r564399934) - I wanted to keep it small and limited. The version of `rum-events-format` which we pin right now doesn't use the notion of dictionary in the JSON schema. 

However, the https://github.com/DataDog/rum-events-format/pull/25 merged recently adds the use of `"additionalProperties"` in JSON schema, which IMO would need to be handled as dictionary. A separate work to support this in `rum-models-generator` is scheduled in `RUMM-1000` JIRA.